client/src/includes/tabs.js

```diff
@@ -0,0 +1,339 @@
+/** Example Markup:
+ <div class="w-tabs" data-tabs data-tabs-animate>
+ <div class="w-tabs__wrapper w-overflow-x-auto u-scrollbar-thin">
+ <div role="tablist" class="w-tabs__list">
+ {% include 'wagtailadmin/shared/tabs/tab_nav_link.html' with tab_id='slug' title='Tab Title' %}
+ </div>
+ <!-- Tab Content -->
+ <section id="tab-{{ tab_id }}" class="w-tabs__panel" role="tabpanel" aria-labelledby="tab-label-{{ tab_id }}" hidden>
+ {{ content }}
+ </section>
+ </div>
+ </div>
+ */
+class Tabs {
+  constructor(node) {
+    this.tabContainer = node;
+    this.tabButtons = this.tabContainer.querySelectorAll('[role="tab"]');
+    this.tabList = this.tabContainer.querySelector('[role="tablist"]');
+    this.tabPanels = this.tabContainer.querySelectorAll('[role="tabpanel"]');
+    this.keydownEventListener = this.keydownEventListener.bind(this);
+
+    // Tab Options - Add these data attributes along side the data-tabs attribute
+    // Use this to enable fade-in animations on tab select
+    this.animate = this.tabContainer.hasAttribute('data-tabs-animate');
+    // Disable url hash from appearing on tab select (normally used in modals)
+    this.disableURL = this.tabContainer.hasAttribute('data-tabs-disable-url');
+
+    this.state = {
+      // Tab Settings
+      activeTabID: '',
+      transition: 150,
+      initialPageLoad: true,
+      // CSS Classes
+      css: {
+        animate: 'animate-in',
+      },
+      // Keyboard Keys
+      keys: {
+        end: 'End',
+        home: 'Home',
+        left: 'ArrowLeft',
+        up: 'ArrowUp',
+        right: 'ArrowRight',
+        down: 'ArrowDown',
+        enter: 'Enter',
+        space: ' ',
+      },
+      direction: {
+        ArrowLeft: -1,
+        Left: -1,
+        ArrowRight: 1,
+        Right: 1,
+      },
+    };
+
+    this.onComponentLoaded();
+  }
+
+  onComponentLoaded() {
+    this.bindEvents();
+
+    // Set active tab from url or make first tab active
+    if (this.tabButtons) {
+      // Set each button's aria-controls attribute and select tab if aria-selected has already been set on the element
+      this.tabButtons.forEach((button) => {
+        button.setAttribute(
+          'aria-controls',
+          button.getAttribute('href').replace('#', ''),
+        );
+      });
+
+      // Check for active items set by the template
+      const tabActive = [...this.tabButtons].find(
+        (button) => button.getAttribute('aria-selected') === 'true',
+      );
+
+      if (window.location.hash && !this.disableURL) {
+        this.selectTabByURLHash();
+      } else if (tabActive) {
+        // If a tab isn't hidden for some reason hide it
+        this.tabPanels.forEach((tab) => {
+          // eslint-disable-next-line no-param-reassign
+          tab.hidden = true;
+        });
+        // Show aria-selected tab
+        this.selectTab(tabActive);
+      } else {
+        this.selectFirstTab();
+      }
+    }
```

Comment:
Can we create an event for tab loaded. CustomEvent - `wagtail:tab-loaded` or similar. 

Dispatched on the tab container?